Warnings from CAMP_EXPAND

I'm getting warnings from CAMP_EXPAND from hipcc 5.2.3.
```
.../RAJA/include/RAJA/pattern/params/forall.hpp:60:7: warning: expression result unused [-Wunused-value]
      CAMP_EXPAND(detail::combine<EXEC_POL>( camp::get<Seq>(f_params.param_tup) ));
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.../RAJA/tpl/camp/include/camp/helpers.hpp:45:3: note: expanded from macro 'CAMP_EXPAND'
  ::camp::detail::__expand_array_type { 0, ((void)(__VA_ARGS__), 0)... }
  ^                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
This can be fixed by naming the array and "sink"ing it.
```
using A = int[];
A {...}; // error unused
```
becomes
```
using A = int[];
A a{...};
camp::sink(a);
```

--- a/include/camp/helpers.hpp
+++ b/include/camp/helpers.hpp
@@ -42,7 +42,8 @@ namespace detail
   using __expand_array_type = int[];
 }
 #define CAMP_EXPAND(...) \
-  ::camp::detail::__expand_array_type { 0, ((void)(__VA_ARGS__), 0)... }
+  ::camp::detail::__expand_array_type camp_unused_expansion_array{ 0, ((void)(__VA_ARGS__), 0)... }; \
+  ::camp::sink(camp_unused_expansion_array)
 
 template <typename Fn, typename... Args>
 CAMP_HOST_DEVICE constexpr void for_each_arg(Fn&& f, Args&&... args)

